addons/account/models/account_journal.py

```diff
@@ -692,18 +692,19 @@ def create_document_from_attachment(self, attachment_ids=None):
             raise UserError(_("No attachment was provided"))
 
         invoices = self.env['account.move']
-        for attachment in attachments:
-            attachment.write({'res_model': 'mail.compose.message'})
-            decoders = self.env['account.move']._get_create_document_from_attachment_decoders()
-            invoice = False
-            for decoder in sorted(decoders, key=lambda d: d[0]):
-                invoice = decoder[1](attachment)
-                if invoice:
-                    break
-            if not invoice:
-                invoice = self.env['account.move'].create({})
-            invoice.with_context(no_new_invoice=True).message_post(attachment_ids=[attachment.id])
-            invoices += invoice
+        with invoice._disable_discount_precision():
```

Comment:
Added docstring
https://github.com/odoo/odoo/blob/b9839c197efe8e1e70b4b3a73f9d9b67a831e725/addons/account/models/account_move.py#L3409-L3420

Reply:
mmh 'kay.. but creating an object with values that you're not supposed to be able to encode might cause problems later on if you edit the document (because you'll loose those values anyway). :man_shrugging:

Reply:
Well that's the thing...
The form view actually ignores those groundings and made it possible to do it that way...

If you encode a discount of `33.33333333` in the form view, for a price of `300.00`
* the total price will be `200.00`
* the discount will be rounded to `33.33` AFTER computing the total price because the ORM trusted the client for the onchange computations

If you encode a discount of `33.33333333` in the create vals, for a price of `300.00`
* the discount will be rounded to `33.33` BEFORE computing the total price (as soon as you set it, because converted to cache value)
* the total price will be `200.01`